apis/v1alpha1/shared_types.go

```diff
@@ -0,0 +1,163 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// All fields in this package are required unless Explicitly marked optional
+// +kubebuilder:validation:Required
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// AdminNetworkPolicyStatus defines the observed state of AdminNetworkPolicy
+type AdminNetworkPolicyStatus struct {
+	Conditions []metav1.Condition `json:"conditions"`
+}
+
+// AdminNetworkPolicySubject defines what resources the policy applies to.
+// Exactly one of the `NamespaceSelector` or `NamespaceAndPodSelector` pointers
+// should be set.
+type AdminNetworkPolicySubject struct {
+	// Namespaces is used to select pods via namespace selectors
+	// +optional
+	Namespaces *metav1.LabelSelector `json:"namespaceSelector,omitempty"`
+	// Pods is used to select pods via namespace AND pod selectors
+	// +optional
+	Pods *NamespacedPodSubject `json:"namespaceAndPodSelector,omitempty"`
+}
+
+// NamespacedPodSubject allows the user to select a given set of pod(s) in
+// selected namespace(s)
+type NamespacedPodSubject struct {
+	// This field follows standard label selector semantics; if empty,
+	// it selects all Namespaces.
+	NamespaceSelector metav1.LabelSelector `json:"namespaceSelector"`
+
+	// Used to explicitly select pods within a namespace; if empty,
+	// it selects all Pods.
+	PodSelector metav1.LabelSelector `json:"podSelector"`
+}
+
+// AdminNetworkPolicyPorts handles selection of traffic based on destination
+// port(s). Exactly one of the fields must be defined.
+type AdminNetworkPolicyPorts struct {
+	// AllPorts cannot be "false" when it is set
+	// AllPorts allows the user to select all ports for all protocols, thus not
+	// selecting traffic based on L4 principles.
+	// If "true" then all ports are selected for all protocols.
+	// +optional
+	AllPorts *bool `json:"allPorts,omitempty"`
+
+	// The list of ports to allow/deny/pass traffic on, each item in this list is
+	// combined using a logical OR. When this field is present it should contain at
+	// least one item, and this rule allows/denies/passes traffic only if the traffic
+	// matches at least one port in the list.
+	// +optional
+	// +kubebuilder:validation:MinItems=1
+	List []AdminNetworkPolicyPort `json:"list,omitempty"`
+}
+
+// AdminNetworkPolicyPort describes a port to select
+type AdminNetworkPolicyPort struct {
+	// The protocol (TCP, UDP, or SCTP) which traffic must match. If not specified, this
+	// field defaults to TCP.
+	// +optional
+	// +kubebuilder:default=TCP
+	Protocol *v1.Protocol `json:"protocol,omitempty"`
+
+	// The port on the given protocol. This can either be a numerical or named
+	// port on a pod, only traffic on the specified protocol AND port will be matched.
+	Port intstr.IntOrString `json:"port"`
```

Comment:
@robscott in Gateway, did we do IntOrString or switcht to `port` and `portName` ?

Reply:
All I see currently in the GW api is a reference to [PortNumber](https://github.com/kubernetes-sigs/gateway-api/blob/master/apis/v1alpha2/shared_types.go#L113)